src/EnergyPlus/UtilityRoutines.hh

```diff
@@ -67,12 +67,12 @@
 #include <EnergyPlus/DataStringGlobals.hh>
 #include <EnergyPlus/DataErrorTracking.hh>
 #include <EnergyPlus/SQLiteProcedures.hh>
+#include <EnergyPlus/Data/EnergyPlusData.hh>
```

Comment:
Be very cautious adding new headers into header files.  Now every file that includes this header file will also thus inherit the entire contents of the data header.  This may be OK, I didn't look at the full context, but a lot of effort has gone into reducing the compile time burden and build dependencies, and this isn't taken lightly.

Reply:
I assume that this is because `ShowSevereError` is now a `template` function and its implementation (which includes dereferences of `state.dataErrTracking`) needs to be in the header file. Bummer. I don't think this dependence can be broken by implementing the state-related functionality via methods on `state`, unless we break `state` into an interface object that contains an implementation object (Microsoft was really big into this pattern back in the day), but that will add a level of indirection everywhere.

Is this problem significant enough that we want to use old-school `va_args` instead of this new-fangled template mechanism?

Reply:
I could not agree more, and this was done with the greatest reluctance. However, the approach of templatizing Show...Error does require a header implementation, which then pulls the whole world along with it. Perhaps we include only the parts of "state" that are required (and maybe that's on the horizon generally), but it makes the error handling function signatures bulkier...

Reply:
Since we are specifically talking about cleaning up the _error_ message function calls for this PR, can we not just pass a vector of strings in and avoid all this templatey stuff?  The simulations about to end, on an error, anyway, so runtime penalties from setting up the vector should not be a concern.

Reply:
To address amir's suggestion, there's a slightly exotic paradigm I was leveraging with the variadic templates in order to support either one or two "default" `InputOutputFile` objects in the parameter list. I don't believe that the default arguments would play nicely with va_args, but I haven't tested that theory yet. (Removing the default arguments might be reasonable, if we can determine they aren't used anywhere!)

Reply:
@Myoldmopar , that was the actual Issue request, you're right. However, it does violate the "principle of least astonishment" for the user of the Error functions, and requires an in-place construction of a vector<string>, if I'm not mistaken.

Reply:
> requires an in-place construction of a vector, if I'm not mistaken.

If you are referring to the performance penalty for constructing a vector, I'm not buying it.  At this point in the simulation we are emitting errors to abort the simulation.  Performance is not a concern.

If you are referring to having to explicitly construct a vector, just use an [initializer list](https://godbolt.org/).  Highlight:
`showError({"foo", "bar"});`

Reply:
I don't think optional arguments play nicely with `va_args`, no. How often are the optional arguments invoked and what are they invoked with? Could they be made non-optional or just eliminated or replaced with some option that lives in `state`?